sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -54,19 +92,23 @@ export class Recorder {
   private variables: Record<string, string>;
   private matcherSet = false;
 
-  constructor(private testContext?: Test | undefined) {
+  constructor(private testContext?: TestInfo) {
+    if (!this.testContext) {
```

Comment:
👍 much prefer this style of coding. Does that also allow us to remove the `if (this.testContext)` call in line 109 since we validated it exists already?

Also, previous code only threw if this.testContext is falsy _and_ we're in recording / playback mode. So should this check also skip throwing if we're in live mode?